internal/winapi/utils.go

```diff
@@ -58,3 +61,21 @@ func ConvertStringSetToSlice(buf []byte) ([]string, error) {
 	}
 	return nil, errors.New("string set malformed: missing null terminator at end of buffer")
 }
+
+// CreateProcThreadAttrList sets up and allocates a process thread attribute list
+// of the right size for use with a STARTUPINFOEX structure.
+func CreateProcThreadAttrList(num uint32) (uintptr, error) {
+	var procThreadSize uintptr
+	if err := InitializeProcThreadAttributeList(0, num, 0, &procThreadSize); err != nil && err != windows.E_NOT_SUFFICIENT_BUFFER {
+		return 0, fmt.Errorf("failed to initialize process thread attribute list: %s", err)
+	}
+
+	buf := make([]byte, procThreadSize)
+	procAttrList := uintptr(unsafe.Pointer(&buf[0]))
```

Comment:
I know kevin has been looking into and concerned that we're doing a lot of unsafe things with pointers and might cause bad problems. This is probably unsafe, just from looking at: 

`
A uintptr is an integer, not a reference. Converting a Pointer to a uintptr creates an integer value with no pointer semantics. Even if a uintptr holds the address of some object, the garbage collector will not update that uintptr's value if the object moves, nor will that uintptr keep the object from being reclaimed.
`

You might also consider this piece of information for the rest of the PR (and I know we probably have a LOT of other instances of these problems so I don't expect those to all get fixed here):

`
If a pointer argument must be converted to uintptr for use as an argument, that conversion must appear in the call expression itself:

syscall.Syscall(SYS_READ, uintptr(fd), uintptr(unsafe.Pointer(p)), uintptr(n))
The compiler handles a Pointer converted to a uintptr in the argument list of a call to a function implemented in assembly by arranging that the referenced allocated object, if any, is retained and not moved until the call completes, even though from the types alone it would appear that the object is no longer needed during the call.
`

Reply:
Yes, this looks like a problem. When GC runs, `uintptr`s aren't considered to be "pointing at" memory for purposes of determining what is still being used, so `buf` could be reclaimed.

Reply:
Yea we spoke about this offline. I can't think of a clean way to do this without HeapAlloc/HeapFree. You want to just go the returning an anonymous function that does the cleanup route?